<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>PhD Book</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="PhD Book">
  <meta name="generator" content="bookdown 0.3.7 and GitBook 2.6.7">

  <meta property="og:title" content="PhD Book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PhD Book" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="r-example-breaks-findhaplotypebreaksmatched-haplotypes-headbreaks.html">
<link rel="next" href="section-2.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="side-projects.html"><a href="side-projects.html"><i class="fa fa-check"></i><b>2</b> Side projects</a><ul>
<li class="chapter" data-level="2.1" data-path="side-projects.html"><a href="side-projects.html#haplotype-project"><i class="fa fa-check"></i><b>2.1</b> Haplotype Project</a><ul>
<li class="chapter" data-level="2.1.1" data-path="side-projects.html"><a href="side-projects.html#background-and-idea"><i class="fa fa-check"></i><b>2.1.1</b> Background and Idea</a></li>
<li class="chapter" data-level="2.1.2" data-path="side-projects.html"><a href="side-projects.html#data"><i class="fa fa-check"></i><b>2.1.2</b> Data</a></li>
<li class="chapter" data-level="2.1.3" data-path="side-projects.html"><a href="side-projects.html#methods"><i class="fa fa-check"></i><b>2.1.3</b> Methods</a></li>
<li class="chapter" data-level="2.1.4" data-path="side-projects.html"><a href="side-projects.html#implementation"><i class="fa fa-check"></i><b>2.1.4</b> Implementation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-example-haplotypes-findconsensusmatched-haplotypes16.html"><a href="r-example-haplotypes-findconsensusmatched-haplotypes16.html"><i class="fa fa-check"></i><b>3</b> <code>{r} # # Example # # haplotypes &lt;- findConsensus(matched) # # haplotypes[1:6] #</code></a></li>
<li class="chapter" data-level="4" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>4</b> </a></li>
<li class="chapter" data-level="5" data-path="r-example-breaks-findhaplotypebreaksmatched-haplotypes-headbreaks.html"><a href="r-example-breaks-findhaplotypebreaksmatched-haplotypes-headbreaks.html"><i class="fa fa-check"></i><b>5</b> <code>{r} # # Example # # breaks &lt;- findHaplotypeBreaks(matched, haplotypes) # # head(breaks) #</code></a></li>
<li class="chapter" data-level="6" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>6</b> </a></li>
<li class="chapter" data-level="7" data-path="section-2.html"><a href="section-2.html"><i class="fa fa-check"></i><b>7</b> </a></li>
<li class="chapter" data-level="8" data-path="r-plot-haplotype.html"><a href="r-plot-haplotype.html"><i class="fa fa-check"></i><b>8</b> <code>{r plot_haplotype} #</code></a></li>
<li class="chapter" data-level="9" data-path="section-3.html"><a href="section-3.html"><i class="fa fa-check"></i><b>9</b> </a></li>
<li class="chapter" data-level="10" data-path="r-findnearestbreaks.html"><a href="r-findnearestbreaks.html"><i class="fa fa-check"></i><b>10</b> <code>{r findNearestBreaks} #</code></a></li>
<li class="chapter" data-level="11" data-path="section-4.html"><a href="section-4.html"><i class="fa fa-check"></i><b>11</b> </a></li>
<li class="chapter" data-level="12" data-path="combining-it-all.html"><a href="combining-it-all.html"><i class="fa fa-check"></i><b>12</b> Combining it all:</a></li>
<li class="chapter" data-level="13" data-path="section-5.html"><a href="section-5.html"><i class="fa fa-check"></i><b>13</b> </a></li>
<li class="chapter" data-level="14" data-path="r-haplotype-mutation.html"><a href="r-haplotype-mutation.html"><i class="fa fa-check"></i><b>14</b> <code>{r haplotype_mutation} #</code></a><ul>
<li class="chapter" data-level="14.0.1" data-path="r-haplotype-mutation.html"><a href="r-haplotype-mutation.html#results"><i class="fa fa-check"></i><b>14.0.1</b> Results</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PhD Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="section-1" class="section level1">
<h1><span class="header-section-number">6</span> </h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">### Map found SNPs to their genomic position <span class="al">###</span>
mapSNPs &lt;-<span class="st"> </span>function(breaks, chr_coords, brca_middle, brca_start, brca_stop, brca_chr){
<span class="co">#mapSNPs &lt;- function(){</span>
    chr_pos =<span class="st"> </span><span class="ot">NULL</span>
    <span class="co"># For each sample (patient) with given mutation</span>
    for (i in <span class="dv">1</span>:<span class="kw">length</span>(breaks)){
        <span class="co"># Extract rows with SNPs in brca1 chromosome (17) and where SNP is homozygote for that sample</span>
        chr_pos2 =<span class="st"> </span><span class="kw">filter</span>(chr_coords, chr_coords$Chr_numeric ==<span class="st"> </span>brca_chr, chr_coords$SNP %in%<span class="st"> </span>breaks[[i]])
        <span class="co"># Removes SNPs in the brca1 gene</span>
        chr_pos2 =<span class="st"> </span><span class="kw">filter</span>(chr_pos2, <span class="kw">as.numeric</span>(<span class="kw">as.character</span>(position_b37)) &lt;<span class="st"> </span>brca_start |<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.character</span>(position_b37)) &gt;<span class="st"> </span>brca_stop)
        <span class="co"># Add sample_id column </span>
        chr_pos2[<span class="st">&quot;sample_id&quot;</span>] =<span class="st"> </span><span class="kw">names</span>(breaks[i])
        <span class="co"># Add position_b37_adjusted column, where brca1 = 0 </span>
        chr_pos2 =<span class="st"> </span><span class="kw">mutate</span>(chr_pos2, <span class="dt">position_b37_adjusted =</span> <span class="kw">as.numeric</span>(<span class="kw">as.character</span>(position_b37)) -<span class="st"> </span>brca_middle)
        <span class="co"># Combine all samples into big data frame for plotting (likely not most efficient way)</span>
        chr_pos =<span class="st"> </span><span class="kw">rbind</span>(chr_pos, chr_pos2)
    }
    <span class="co"># Add columns Country and FamCode to data frame</span>
    index &lt;-<span class="st"> </span><span class="kw">fmatch</span>(chr_pos$sample_id, brca_data_temp$Onc_ID)
    chr_pos[<span class="kw">c</span>(<span class="st">&quot;Country&quot;</span>, <span class="st">&quot;FamCode&quot;</span>)] &lt;-<span class="st"> </span>brca_data_temp[index, ] %&gt;%<span class="st"> </span><span class="kw">select</span>(Country, FamCode)
    
    <span class="co"># Print some data information - here counting samples by Country</span>
    chr_pos %&gt;%<span class="st"> </span><span class="kw">distinct</span>(sample_id, FamCode, Country) %&gt;%<span class="st"> </span><span class="kw">count</span>(Country)
    <span class="kw">dim</span>(chr_pos)
    <span class="co"># unique(chr_pos$sample_id)</span>
    <span class="co"># print(chr_pos)</span>
    <span class="kw">return</span>(chr_pos)    
}</code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="r-example-breaks-findhaplotypebreaksmatched-haplotypes-headbreaks.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="section-2.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
