<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>PhD Book</title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="PhD Book">
  <meta name="generator" content="bookdown 0.3.7 and GitBook 2.6.7">

  <meta property="og:title" content="PhD Book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="PhD Book" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="introduction.html">


<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="side-projects.html"><a href="side-projects.html"><i class="fa fa-check"></i><b>2</b> Side projects</a><ul>
<li class="chapter" data-level="2.1" data-path="side-projects.html"><a href="side-projects.html#haplotype-project"><i class="fa fa-check"></i><b>2.1</b> Haplotype Project</a><ul>
<li class="chapter" data-level="2.1.1" data-path="side-projects.html"><a href="side-projects.html#background-and-idea"><i class="fa fa-check"></i><b>2.1.1</b> Background and Idea</a></li>
<li class="chapter" data-level="2.1.2" data-path="side-projects.html"><a href="side-projects.html#data"><i class="fa fa-check"></i><b>2.1.2</b> Data</a></li>
<li class="chapter" data-level="2.1.3" data-path="side-projects.html"><a href="side-projects.html#methods"><i class="fa fa-check"></i><b>2.1.3</b> Methods</a></li>
<li class="chapter" data-level="2.1.4" data-path="side-projects.html"><a href="side-projects.html#implementation"><i class="fa fa-check"></i><b>2.1.4</b> Implementation</a></li>
<li class="chapter" data-level="2.1.5" data-path="side-projects.html"><a href="side-projects.html#results"><i class="fa fa-check"></i><b>2.1.5</b> Results</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PhD Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="side-projects" class="section level1">
<h1><span class="header-section-number">2</span> Side projects</h1>
<p>Topics in this category describes some of the side projects I have been during - related to but not part of - my PhD project.</p>

<div id="haplotype-project" class="section level2">
<h2><span class="header-section-number">2.1</span> Haplotype Project</h2>
<p><em>Updated: 15-01-2017</em></p>
<div id="background-and-idea" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Background and Idea</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="dv">2+2</span>
<span class="co">#&gt; [1] 4</span></code></pre></div>
</div>
<div id="data" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Data</h3>
<p>This section describes the raw data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Read onco phenotype data</span>
brca_data &lt;-<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;Onco_pheno_BC1_Example.xlsx&quot;</span>)
brca_data[, <span class="dv">1</span>:<span class="dv">10</span>]
<span class="co">#&gt; # A tibble: 8 × 10</span>
<span class="co">#&gt;   FamCode Onc_ID     Country Ascertainment Mut1Gene  Mut1BIC   Mut1HGVS</span>
<span class="co">#&gt;     &lt;dbl&gt;  &lt;dbl&gt;       &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;    &lt;chr&gt;      &lt;chr&gt;</span>
<span class="co">#&gt; 1       1      1       ITALY             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 2       2      2       ITALY             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 3       3      3 NETHERLANDS             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 4       3      4 NETHERLANDS             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 5       4      5         USA             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 6       4      6         USA             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 7       5      7     AUSTRIA             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; 8       6      8     AUSTRIA             1        1 1135delA c.1016delA</span>
<span class="co">#&gt; # ... with 3 more variables: Mut1HGVS_Protein &lt;chr&gt;, Mut1Type &lt;chr&gt;,</span>
<span class="co">#&gt; #   Mut1Effect &lt;chr&gt;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># read SNP information</span>
haplotypes_brca &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;haplotypes_brca1_example.txt&quot;</span>, <span class="dt">header =</span> T)
haplotypes_brca[<span class="dv">1</span>:<span class="dv">10</span>, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">10</span>,<span class="dv">15</span>, <span class="dv">55</span>)]
<span class="co">#&gt;    SNP X1KG_13_37679380 c13_pos31809263 c13_pos31826936 c17_pos50397016</span>
<span class="co">#&gt; 1   16               DD              AA              AA              GG</span>
<span class="co">#&gt; 2   12               DD              AA              AA              GG</span>
<span class="co">#&gt; 3  133               DD              AA              AA              CG</span>
<span class="co">#&gt; 4  162               DD              AA              AA              GG</span>
<span class="co">#&gt; 5  142               DD              AA              AA              CG</span>
<span class="co">#&gt; 6  262               DD              AA              AA              CG</span>
<span class="co">#&gt; 7  492               DD              AA              AA              CC</span>
<span class="co">#&gt; 8  302               DD              AA              AA              GG</span>
<span class="co">#&gt; 9   65               DD              AA              AA              CG</span>
<span class="co">#&gt; 10  42               DD              AA              AA              GG</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Read SNP location information</span>
chr_coords &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;qry_139_ouh_geno_snps_with_position.txt&quot;</span>, <span class="dt">header =</span> T, <span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)
<span class="kw">head</span>(chr_coords)
<span class="co">#&gt;                SNP Chr_numeric position_b37</span>
<span class="co">#&gt; 1 X1KG_13_37679380          13     37679380</span>
<span class="co">#&gt; 2 X1KG_17_40911775          17     40911775</span>
<span class="co">#&gt; 3 X1KG_17_45479497          17     45479497</span>
<span class="co">#&gt; 4  c13_pos27783810          13     28885810</span>
<span class="co">#&gt; 5  c13_pos31788625          13     31788625</span>
<span class="co">#&gt; 6  c13_pos31791426          13     32893426</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Define BRCA1 gene information</span>
brca_start &lt;-<span class="st"> </span><span class="dv">41197695</span>
brca_stop &lt;-<span class="st"> </span><span class="dv">41276113</span>
brca_middle &lt;-<span class="st"> </span>(brca_stop -<span class="st"> </span>brca_start)/<span class="dv">2</span> +<span class="st"> </span>brca_start
brca_chr &lt;-<span class="st"> </span><span class="dv">17</span></code></pre></div>
</div>
<div id="methods" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Methods</h3>
<ol style="list-style-type: decimal">
<li><code>Filter out SNPs for relating to the chromosome we're working on</code></li>
</ol>
</div>
<div id="implementation" class="section level3">
<h3><span class="header-section-number">2.1.4</span> Implementation</h3>
<p>The <code>extractSamples</code> method…</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">extractSamples &lt;-<span class="st"> </span>function(brca_data, haplotypes_brca, <span class="dt">distinct =</span> F){
    <span class="st">&quot;</span>
<span class="st">    Extract relevant samples from table of all genotypes </span>
<span class="st">    &quot;</span>
    
    <span class="co"># Keep only one member of a family and remove the others, if chosen</span>
    if (distinct){
        brca_data &lt;-<span class="st"> </span>brca_data[!<span class="kw">duplicated</span>(brca_data$FamCode), ]
        <span class="co"># Equivalent to above</span>
        <span class="co"># brca_data &lt;- subset(brca_data, !duplicated(FamCode)) </span>
        <span class="co"># brca_data &lt;- distinct(brca_data, FamCode, .keep_all = T)</span>
    }
    
    <span class="co"># Extract samples (with given brca mutation) from the big haplotypes dataset</span>
    <span class="co"># Faster to use fmatch than match</span>
    index &lt;-<span class="st"> </span><span class="kw">fmatch</span>(brca_data$Onc_ID, haplotypes_brca$SNP)
    matched &lt;-<span class="st"> </span>haplotypes_brca[index, ]
    
    <span class="co"># as above, but slower</span>
    <span class="co"># matched &lt;- filter(haplotypes_brca, SNP %in% brca_data$Onc_ID) </span>
    
    <span class="kw">return</span>(matched)
}</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Example</span>
matched &lt;-<span class="st"> </span><span class="kw">extractSamples</span>(brca_data, haplotypes_brca, <span class="ot">FALSE</span>)
matched[<span class="dv">1</span>:<span class="dv">10</span>, <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">10</span>,<span class="dv">15</span>, <span class="dv">55</span>)]
<span class="co">#&gt;      X1KG_13_37679380 c13_pos31809263 c13_pos31826936 c17_pos50397016</span>
<span class="co">#&gt; 2572               DD              AA              AA              CG</span>
<span class="co">#&gt; 2586               DD              AA              AA              CG</span>
<span class="co">#&gt; 2600               DD              AA              AA              GG</span>
<span class="co">#&gt; 2573               DD              AA              AA              GG</span>
<span class="co">#&gt; 2580               DD              AA              AA              GG</span>
<span class="co">#&gt; 2587               DD              AA              AA              GG</span>
<span class="co">#&gt; 2592               DD              AA              AA              GG</span>
<span class="co">#&gt; 2596               DD              AA              AA              CG</span>
<span class="co">#&gt; 2564               DD              AA              AA              CG</span>
<span class="co">#&gt; 2574               DD              AA              AA              GG</span>
<span class="kw">dim</span>(matched)
<span class="co">#&gt; [1] 15679 12468</span></code></pre></div>
<!-- #```{r findConsensus} -->
<!-- #``` -->
<!-- # ```{r} -->
<!-- # # Example -->
<!-- # # haplotypes <- findConsensus(matched) -->
<!-- # # haplotypes[1:6] -->
<!-- # ``` -->
<!-- #  -->
<!-- Good to see you -->
<!-- #```{r findHaplotypeBreaks} -->
<!-- #``` -->
<!-- # ```{r} -->
<!-- # # Example -->
<!-- # # breaks <- findHaplotypeBreaks(matched, haplotypes) -->
<!-- # # head(breaks) -->
<!-- # ``` -->
<!-- #  -->
<!-- #```{r mapSNPs} -->
<!-- #``` -->
<!-- #  -->
<!-- # ```{r plot_haplotype} -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r findNearestBreaks} -->
<!-- # ``` -->
<!-- #  -->
<!-- # Combining it all: -->
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">### Haplotypes samples with a given brca mutation <span class="al">###</span>
haplotype_mutation &lt;-<span class="st"> </span>function(brca_data, haplotypes_brca, chr_coords, 
                               brca_start, brca_stop, brca_middle, brca_chr, 
                               mut, <span class="dt">distinct =</span> F){
    <span class="co"># TODO</span>
    <span class="co">#   - allow for country to define haplotype_consensus</span>
    
    <span class="co"># Prints for checking output</span>
    <span class="kw">print</span>(<span class="kw">dim</span>(brca_data)) <span class="co"># 8x67</span>
    <span class="kw">print</span>(<span class="kw">dim</span>(haplotypes_brca)) <span class="co"># 15679x12468</span>
    <span class="kw">print</span>(<span class="kw">dim</span>(chr_coords)) <span class="co"># 12467x3</span>
    <span class="kw">print</span>(mut)
    
    <span class="co"># Make brca_data a global variable (for testing and debugging)</span>
    brca_data_temp &lt;&lt;-<span class="st"> </span>brca_data
    brca_data &lt;&lt;-<span class="st"> </span>brca_data_temp
    
    
    matched &lt;&lt;-<span class="st"> </span><span class="kw">extractSamples</span>(brca_data, haplotypes_brca, distinct)
    
    chr_coords2 =<span class="st"> </span><span class="kw">filter</span>(chr_coords, chr_coords$Chr_numeric ==<span class="st"> </span>brca_chr)
    matched &lt;-<span class="st"> </span><span class="kw">select</span>(matched, <span class="kw">na.omit</span>(<span class="kw">fmatch</span>(<span class="kw">colnames</span>(matched[<span class="dv">2</span>:<span class="kw">ncol</span>(matched)]), chr_coords2$SNP)))
    <span class="co"># matched &lt;- matched[, na.omit(fmatch(colnames(matched[2:ncol(matched)]), chr_coords2$SNP))]</span>
    
    haplotypes &lt;&lt;-<span class="st"> </span><span class="kw">findConsensus</span>(matched)
    <span class="co"># haplotypes &lt;&lt;- findConsensus(filter(matched, SNP %in% subset(brca_data, Country == &quot;SPAIN&quot;)$Onc_ID))</span>
    
    breaks &lt;&lt;-<span class="st"> </span><span class="kw">findHaplotypeBreaks</span>(matched, haplotypes)
    
    if (<span class="kw">length</span>(breaks) ==<span class="st"> </span><span class="dv">0</span>){
        <span class="kw">print</span>(<span class="kw">paste</span>(<span class="st">&quot;No haplotype breaks found in families for mutation:&quot;</span>, mut))
        <span class="kw">return</span>()
    }
    
    chr_pos &lt;&lt;-<span class="st"> </span><span class="kw">mapSNPs</span>(breaks, chr_coords, brca_middle, brca_start, brca_stop, brca_chr)
    <span class="co">#chr_pos &lt;- mapSNPs(breaks, chr_coords, brca1_middle, brca1_start, brca1_stop)</span>
    <span class="co">#chr_pos &lt;- mapSNPs()</span>
    <span class="kw">save</span>(chr_pos, <span class="dt">file =</span> <span class="kw">paste0</span>(<span class="st">&quot;chr_pos-&quot;</span>, brca_name, <span class="st">&quot;-&quot;</span>, <span class="kw">paste</span>(mut, <span class="dt">collapse =</span> <span class="st">&quot;,&quot;</span>), <span class="st">&quot;.RData&quot;</span>))
    
    <span class="kw">plot_haplotype</span>(chr_pos, brca_chr, mut)
    
    <span class="kw">findNearestBreaks</span>(chr_pos)
}</code></pre></div>
</div>
<div id="results" class="section level3">
<h3><span class="header-section-number">2.1.5</span> Results</h3>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>


<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
